src/map/status.c

```diff
@@ -1560,6 +1560,44 @@ static int status_fixed_revive(struct block_list *bl, unsigned int per_hp, unsig
 	return 1;
 }
 
+/**
+ * a sub-function from 'status_check_skilluse' function, which can be use for plugin manipulation
+ * check 'disabled_skills' flag settings from `db\(pre-)re\map_zone_db.conf`
+ *
+ * @param src        The source object can be BL_PC, BL_MOB, BL_PET, BL_HOM, BL_MER, BL_ELEM type
+ * @param st         The status_data of the source object, use to check MD_BOSS of BL_MOB type
+ * @param skill_id   The skill ID use by src
+ *
+ * @return false if the skill ID use is denied, true otherwise
+ **/
+static bool status_check_skilluse_mapzone(struct block_list *src, struct status_data *st, uint16 skill_id)
+{
+	nullpo_retr(false, src);
+	nullpo_retr(false, st);
+	for (int i = 0; i < map->list[src->m].zone->disabled_skills_count; i++) {
+		if (skill_id == map->list[src->m].zone->disabled_skills[i]->nameid && (map->list[src->m].zone->disabled_skills[i]->type & src->type) != 0) {
```

Comment:
This line is way too long. (156 columns.)
Since keeping the conditions in a single line doesn't significantly increases the readability break it please.